### ISSUE TITLE
Random crash on Kobo Glo

### ISSUE THREAD

**Issue by X-static** [koreader/koreader#10302]
* KOReader version: Koreader v2023.03
* Device: Kobo Glo

#### Issue
I'm getting random crashes on latest version. Seems to be something 
/on kosync plugin.

`plugins/kosync.koplugin/main.lua:550: in function 'getDocumentDigest'
	plugins/kosync.koplugin/main.lua:633: in function 'getProgress'
	plugins/kosync.koplugin/main.lua:784: in function 'action'
	frontend/ui/uimanager.lua:980: in function '_checkTasks'
	frontend/ui/uimanager.lua:1426: in function 'handleInput'
	frontend/ui/uimanager.lua:1549: in function 'run'
	./reader.lua:319: in main chunk`

#### Steps to reproduce
Couldn't find a pattern. It happened after waking the device from sleep and also while reading.

##### `crash.log` (if applicable)
`crash.log` is a file that is automatically created when KOReader crashes. It can normally be found in the KOReader directory:

* `/mnt/private/koreader` for Cervantes
* `koreader/` directory for Kindle
* `.adds/koreader/` directory for Kobo
* `applications/koreader/` directory for Pocketbook

Android logs are kept in memory. Please go to [Menu] → Help → Bug Report to save these logs to a file.

Please try to include the relevant sections in your issue description.
You can upload the whole `crash.log` file (zipped if necessary) on GitHub by dragging and dropping it onto this textbox.

If your issue doesn't directly concern a Lua crash, we'll quite likely need you to reproduce the issue with *verbose* debug logging enabled before providing the logs to us.
To do so, from the file manager, go to [Tools] → More tools → Developer options, and tick both `Enable debug logging` and `Enable verbose debug logging`.
You'll need to restart KOReader after toggling these on.

If you instead opt to inline it, please do so behind a spoiler tag:
<details>
  <summary>crash.log</summary>

```
<Paste crash.log content here>
```
#10223 
[crash.log](https://github.com/koreader/koreader/files/11173702/crash.log)

**Comment by NiLuJe** [comment]
Stupid question is stupid, but one never knows: do you actually use KOSync and/or have it setup?

**Comment by NiLuJe** [comment]
Okay, looking at the logs, I would assume the answer to be yes.

**Comment by NiLuJe** [comment]
Actual uncropped traceback:

```
./luajit: plugins/kosync.koplugin/main.lua:550: attempt to index field 'document' (a nil value)
stack traceback:
	plugins/kosync.koplugin/main.lua:550: in function 'getDocumentDigest'
	plugins/kosync.koplugin/main.lua:633: in function 'getProgress'
	plugins/kosync.koplugin/main.lua:784: in function 'action'
	frontend/ui/uimanager.lua:980: in function '_checkTasks'
	frontend/ui/uimanager.lua:1426: in function 'handleInput'
	frontend/ui/uimanager.lua:1549: in function 'run'
	./reader.lua:319: in main chunk
	[C]: at 0x000140bd
```

Possibly related to an early `onNetworkConnected` event, given the timing?

**Comment by X-static** [comment]
> Stupid question is stupid, but one never knows: do you actually use KOSync and/or have it setup?

Yes. I do. I use it to sync reading position with an Android phone.
